nixos/modules/services/continuous-integration/buildkite-agent.nix

```diff
@@ -219,7 +219,7 @@ in
             cat > "${cfg.dataDir}/buildkite-agent.cfg" <<EOF
             token="$(cat ${toString cfg.tokenPath})"
             name="${cfg.name}"
-            meta-data="${cfg.meta-data}"
+            tags="${cfg.tags}"
```

Comment:
It looks like this change was introduced in https://github.com/buildkite/agent/blob/master/CHANGELOG.md#added-18

The default version the module uses is 2.6.1. Will this not fail given tags instead of meta-data?

A solution would be to either support both or at the very least change the default and assert that the version is in fact above 3.0-beta.22

Reply:
Seems like unnecessary complication to me. Do we really need to support 2.6.1 given that 3.0 has been stable for months? At work we found that <3.0 in nixpkgs didn't even work correctly, and required patches to run at all.

Reply:
What would you suggest instead? In the current state of this PR, enabling the agent would simply fail unless the package is overwritten.

As I said in my second option, I think it's fine to make this module incompatible with earlier versions, but then it should in fact be made incompatible with earlier versions. At the very least, the default package should change, and a release note would be great, if relevant. :)

Reply:
Alright, I agree that if both versions are present and available in nixpkgs, both should be usable by the available service module. I was postulating we should deprecate the older version, but I'm not sure I should be doing that in this PR. Which makes this PR kinda pointless. I'll try to submit a new PR that addresses my position correctly.